packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts

```diff
@@ -152,13 +169,60 @@ export const createRdsLambda = (
       ),
     ],
     lambdaRole,
-    environment,
+    {
+      ...environment,
+      SSM_ENDPOINT: ssmEndpoint,
+    },
     Duration.seconds(30),
     scope,
-    sqlLambdaVpcConfig,
+    sqlLambdaVpcConfig?.vpcConfig,
   );
 };
 
+const addVpcEndpoint = (scope: Construct, sqlLambdaVpcConfig: VpcSubnetConfig, serviceSuffix: string): CfnVPCEndpoint => {
+  const serviceEndpointPrefix = 'com.amazonaws';
+  return new CfnVPCEndpoint(scope, `RDSVpcEndpoint${serviceSuffix}`, {
+    serviceName: Fn.join('', [serviceEndpointPrefix, '.', Fn.ref('AWS::Region'), '.', serviceSuffix]), // Sample: com.amazonaws.us-east-1.ssmmessages
+    vpcEndpointType: 'Interface',
+    vpcId: sqlLambdaVpcConfig.vpcConfig.vpcId,
+    subnetIds: extractSubnetForVpcEndpoint(sqlLambdaVpcConfig.subnetAvailabilityZoneConfig),
+    securityGroupIds: sqlLambdaVpcConfig.vpcConfig.securityGroupIds,
+    privateDnsEnabled: false,
+  });
+};
+
+const addVpcEndpointForSecretsManager = (
```

Comment:
naming nit: You're setting up more endpoints than just secretsmanager here

Reply:
Though we are creating endpoints for different services, all these are required for secrets manager to work E2E.

Refer here:
https://docs.aws.amazon.com/systems-manager/latest/userguide/setup-create-vpc.html#sysman-setting-up-vpc-create